embrace-gradle-plugin/src/main/java/io/embrace/android/gradle/plugin/EmbraceLogger.kt

```diff
@@ -0,0 +1,51 @@
+@file:Suppress("Filename", "MatchingDeclarationName")
+
+package io.embrace.android.gradle.plugin
+
+import org.gradle.api.logging.Logging
+
+/**
+ * Embrace-specific logger that automatically adds plugin and class prefixes.
+ */
+class EmbraceLogger {
+    private val gradleLogger: org.gradle.api.logging.Logger
+    private val logPrefix: String
+
+    constructor(clazz: Class<*>) {
+        gradleLogger = Logging.getLogger(clazz)
+        logPrefix = "[EmbraceGradlePlugin] [${clazz.simpleName}]"
+    }
+
+    constructor(componentName: String) {
+        gradleLogger = Logging.getLogger(componentName)
+        logPrefix = "[EmbraceGradlePlugin] [$componentName]"
+    }
```

Comment:
nit: suggest a primary constructor with a `componentName` parameter that these secondary constructors call, rather than duplicating code